agent/xds/response.go

```diff
@@ -0,0 +1,54 @@
+package xds
+
+import (
+	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoycore "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+)
+
+func createResponse(typeURL string, version, nonce string, resources []proto.Message) (*envoy.DiscoveryResponse, error) {
+	anys := make([]types.Any, len(resources))
```

Comment:
I think because you have `continue` skips in the loop below, you should switch this to `make([]types.Any, 0, len(resources)` and use `append` so you don't have `nil` landmines in slice.

Reply:
Good call, The nil check in the loop was added later and happens to only happen so far if for some reason there is a single nil result so wasn't causing issues but this def is a bug waiting to happen. Will change.